.github/workflows/release_snapshot.yml

```diff
@@ -52,69 +51,47 @@ jobs:
            --operation=create
           source ./result.env
           echo "$CREATED_TAG" 
-          echo "created_tag=$CREATED_TAG" >> $GITHUB_OUTPUT
-  linux:
-    needs: linux-create
-    runs-on: ubuntu-latest
+          echo "created_tag=$CREATED_TAG" >> "$GITHUB_OUTPUT"
+          echo "created_tag=$CREATED_TAG" >> $GITHUB_ENV
+      - uses: ./.github/actions/test_install
+        with:
+          version_number: ${{ env.created_tag }}
+          python_version: "3.8"
+          os_platform: "linux"
+  build-for-os-and-python-versions:
+    needs: create-snapshot
     strategy:
       matrix:
         python-version: ["3.7", "3.9", "3.10"]
+        os: ["macos-latest", "ubuntu-latest"]
+        include:
+          - python-version: 3.8
+            os: macos-latest
     env:
       GH_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-      - run: |
+      - if: matrix.os == 'ubuntu-latest'
+        run: |
+          echo "install_os=linux" >> $GITHUB_ENV
           sudo apt-get install libsasl2-dev
+      - if: matrix.os == 'macos-latest'
+        run: |
+          echo "install_os=mac" >> $GITHUB_ENV
+      - run: |    
           python -m pip install --user --upgrade pip
           pip install -r requirements.txt
       - run: |
           python release_creation/main.py \
            --input-version=${{ inputs.version_number }} \
            --operation=update
-  macos:
-    needs: linux-create
-    runs-on: macos-latest
-    strategy:
-      matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
-    env:
-      GH_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-      - run: |
-          python -m pip install --user --upgrade pip
-          pip install -r requirements.txt
-      - run: |
-          python release_creation/main.py \
-           --input-version=${{ inputs.version_number }} \
-           --operation=update
-  test-linux-install:
-    needs: linux-create
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set up Python 3.8
-        uses: actions/setup-python@v4
+      - uses: ./.github/actions/test_install
```

Comment:
Why did you write this as an action instead of a reusable workflow?  I tend to shy away from composite actions mostly because of the lack of logging out of it.

Reply:
Performance wise it seems more efficient since it can run on the same host. Also as we add more versions and platforms we could end up exceeding the 20 concurrent workflow cap

Reply:
If there's no value in getting any logging out of why the install test fails that's fine.  You could also pass a reason as output and just manually log it.

As for the 20 workflow limit, I read that as unique workflows as opposed to this setup with one workflow repeated as part of a matrix.  That would need a test to prove it out though.